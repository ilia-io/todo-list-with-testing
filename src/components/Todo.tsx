import React, { useState } from 'react';

export interface ITodo {
  id: number;
  title: string;
  isCompleted: boolean;
}

interface ITodoProps {
  id: number;
  title: string;
  isCompleted: boolean;
  deleteTodo: (id: number) => void;
  updateCheckStatus: (id: number) => void;
}
const Todo: React.FC<ITodoProps> = ({
  id,
  title,
  isCompleted,
  deleteTodo,
  updateCheckStatus,
}) => {
  const [completed, setCompleted] = useState(isCompleted);

  const onChangeStatus = () => {
    setCompleted(!completed);
    updateCheckStatus(id);
  };

  return (
    <li data-testid="list-item">
      <input
        data-testid="toggle-checkbox"
        onChange={onChangeStatus}
        checked={completed}
        type="checkbox"
      />
      <span
        data-testid="toggle-text"
        style={
          isCompleted
            ? {
                textDecorationLine: 'line-through',
                textDecorationThickness: '1px',
              }
            : {}
        }
      >
        {title}
      </span>
      <button
        data-testid="list-btn"
        onClick={() => deleteTodo(id)}
        type="button"
      >
        x
      </button>
    </li>
  );
};

export default Todo;
