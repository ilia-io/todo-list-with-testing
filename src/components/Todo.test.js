import { fireEvent, render, screen } from '@testing-library/react';
import Todo from './Todo';

describe('Todo component tests', () => {
  const data = { title: 'func', isCompleted: true };

  it('List renders', () => {
    render(<Todo title={data.title} />);
    const li = screen.getByRole('listitem');
    const input = screen.getByRole('checkbox');
    const title = screen.getByText('func');
    expect(li).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(title).toBeInTheDocument();
    expect(li).toMatchSnapshot();
  });

  it('Props text line-t', () => {
    render(<Todo isCompleted={true} title={'test'} />);
    expect(screen.getByTestId('toggle-text')).toHaveStyle({
      textDecorationLine: 'line-through',
    });
  });
  it('Props text line-none', () => {
    render(<Todo isCompleted={false} title={'test'} />);
    expect(screen.getByTestId('toggle-text')).not.toHaveAttribute('style');
  });
  it('delete button function', () => {
    let testId = 0;
    render(<Todo id={22} deleteTodo={(id) => (testId = id)} />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    fireEvent.click(btn);
    expect(testId).toBe(22);
  });
});
