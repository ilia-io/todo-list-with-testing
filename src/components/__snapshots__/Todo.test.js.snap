// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component tests List renders 1`] = `
<li
  data-testid="list-item"
>
  <input
    data-testid="toggle-checkbox"
    type="checkbox"
  />
  <span
    data-testid="toggle-text"
  >
    func
  </span>
  <button
    data-testid="list-btn"
    type="button"
  >
    x
  </button>
</li>
`;
